packages/core/src/actions/execute.ts

```diff
@@ -496,19 +696,16 @@ export const executeActionsViaSigner: ExecuteActions = async (
     overrides.gasLimit = limit
   }
 
-  const txn = await getGasPriceOverrides(provider, signer, executionMode, {
-    to: moduleAddress,
-    data: executionData,
-    ...overrides,
-  })
-  const receipt = await (await signer.sendTransaction(txn)).wait()
-
-  // Narrow the TypeScript type.
-  if (!receipt) {
-    throw new Error(`Receipt is null. Should never happen.`)
-  }
-
-  return receipt
+  return deploymentContext.executeTransaction(
+    deploymentContext,
+    {
+      to: moduleAddress,
+      data: executionData,
+      gasLimit: overrides.gasLimit?.toString(),
+      chainId: (await provider.getNetwork()).chainId.toString(),
+    },
+    executionMode
+  )
```

Comment:
It's not strictly necessary to use the `deploymentContext.executeTransaction` function here since `executeActionsViaSigner` won't get called on the website. However, I like to make things easy to reason about so in this PR I tried to follow the general rule of "if the transaction is valid on a live network, then use executeTransaction". I find that rule to be easier to reason about than trying to think about whether each specific transaction will need to be executed on the website or not. 

For this reason, I recommend we generally bias towards always using the deploymentContext.executeTransaction function even if it's not always strictly necessary. That being said, this PR doesn't fully follow that anyway (ensureSphinxAndGnosisSafeDeployed doesn't use it for example). 

I mainly think it will save you time and mental effort to just use this function all the time since you don't really work on the website backend at all.